corehq/apps/api/odata/tests/test_feed.py

```diff
@@ -142,3 +143,123 @@ def test_request_succeeded(self):
         correct_credentials = self._get_correct_credentials()
         response = self._execute_query(correct_credentials)
         self.assertEqual(response.status_code, 200)
+
+
+class TestFormOdataFeed(TestCase, FormOdataTestMixin):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestFormOdataFeed, cls).setUpClass()
+
+        cls.client = Client()
+        cls.domain = Domain(name='test_domain')
+        cls.domain.save()
+        cls.web_user = WebUser.create(cls.domain.name, 'test_user', 'my_password')
+
+        cls.account, _ = BillingAccount.get_or_create_account_by_domain(cls.domain.name, created_by='')
+        plan_version = DefaultProductPlan.get_default_plan_version(SoftwarePlanEdition.STANDARD)
+        cls.subscription = Subscription.new_domain_subscription(cls.account, cls.domain.name, plan_version)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain.delete()
+        cls.web_user.delete()
+
+        SubscriptionAdjustment.objects.all().delete()
+        cls.subscription.delete()
+        cls.account.delete()
+        super(TestFormOdataFeed, cls).tearDownClass()
+
+    def test_no_credentials(self):
+        response = self.client.get(self.view_url)
+        self.assertEqual(response.status_code, 404)
+
+    def test_wrong_password(self):
+        wrong_credentials = self._get_basic_credentials(self.web_user.username, 'wrong_password')
+        response = self._execute_query(wrong_credentials)
+        self.assertEqual(response.status_code, 404)
+
+    def test_wrong_domain(self):
+        other_domain = Domain(name='other_domain')
+        other_domain.save()
+        self.addCleanup(other_domain.delete)
+
+        correct_credentials = self._get_correct_credentials()
+        response = self.client.get(
+            self._odata_feed_url_by_domain(other_domain.name),
+            HTTP_AUTHORIZATION='Basic ' + correct_credentials,
+        )
+        self.assertEqual(response.status_code, 404)
+
+    def test_missing_feature_flag(self):
+        correct_credentials = self._get_correct_credentials()
+        response = self._execute_query(correct_credentials)
+        self.assertEqual(response.status_code, 404)
+
+    @flag_enabled('ODATA')
+    def test_request_succeeded(self):
+        with trap_extra_setup(ConnectionError):
+            elasticsearch_instance = get_es_new()
+            initialize_index_and_mapping(elasticsearch_instance, XFORM_INDEX_INFO)
+        self.addCleanup(self._ensure_case_index_deleted)
+
+        self.web_user.set_role(self.domain.name, 'admin')
+        self.web_user.save()
+
+        correct_credentials = self._get_correct_credentials()
+        response = self._execute_query(correct_credentials)
+        self.assertEqual(response.status_code, 200)
```

Comment:
should the data/format of the response be tested here?

Reply:
121db162e804adeaf33bf01045bf11a1d45f6966